lib/msf/core/post/linux/compile.rb

```diff
@@ -36,6 +36,16 @@ def live_compile?
   def upload_and_compile(path, data, compiler_args='')
     write_file "#{path}.c", strip_comments(data)
 
+    if datastore['COMPILER'] == 'Auto'
+      if has_gcc?
+        datastore['COMPILER'] = 'gcc'
+      elsif has_clang?
+        datastore['COMPILER'] = 'clang'
+      else
+        fail_with Module::Failure::BadConfig, "Unable to find gcc nor clang on the remote target."
+      end
+    end
+
```

Comment:
Clobbering `datastore` values is generally frowned upon.

Can we do this with a `compiler` variable within this method instead?

I'm not aware of any existing scenario where the calling module (or anything else outside of `upload_and_compile`) will need to know what compiler was used.